src/main/java/io/quarkus/search/app/hibernate/GuideLoadingBinder.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.search.app.hibernate;
+
+import java.net.URI;
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Named;
+
+import io.quarkus.search.app.entity.Guide;
+
+import org.hibernate.search.mapper.pojo.standalone.loading.LoadingTypeGroup;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassEntityLoader;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassEntitySink;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierLoader;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierSink;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassLoadingOptions;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassLoadingStrategy;
+import org.hibernate.search.mapper.pojo.standalone.loading.binding.EntityLoadingBinder;
+import org.hibernate.search.mapper.pojo.standalone.loading.binding.EntityLoadingBindingContext;
+
+@ApplicationScoped
+@Named("guideLoadingBinder")
+public class GuideLoadingBinder implements EntityLoadingBinder {
+
+    @Override
+    public void bind(EntityLoadingBindingContext entityLoadingBindingContext) {
+        entityLoadingBindingContext.massLoadingStrategy(Guide.class, new MassLoadingStrategy<Guide, URI>() {
+            @Override
+            public boolean equals(Object obj) {
+                return false;
+            }
+
+            @Override
+            public int hashCode() {
+                return 0;
+            }
+
+            @Override
+            public MassIdentifierLoader createIdentifierLoader(LoadingTypeGroup<Guide> includedTypes,
+                    MassIdentifierSink<URI> sink, MassLoadingOptions options) {
+                QuarkusIOLoadingContext context = options.context(QuarkusIOLoadingContext.class);
+                return new MassIdentifierLoader() {
+                    @Override
+                    public void close() {
+                    }
+
+                    @Override
+                    public long totalCount() {
+                        return context.size();
+                    }
+
+                    @Override
+                    public void loadNext() throws InterruptedException {
+                        List<URI> batch = context.nextIdBatch(options.batchSize());
+                        if (batch.isEmpty()) {
+                            sink.complete();
+                        } else {
+                            sink.accept(batch);
+                        }
+                    }
+                };
+            }
```

Comment:
Did you consider having the identifier loader return the `Guide` instances directly?

Your strategy would simply implement `MassLoadingStrategy<Guide, Guide>`, and your entity loader would be a no-op (there's nothing to load).

Reply:
ha 😃 I was thinking id is id, silly me 😄. I've updated to a `<Guide,Guide>` variant and all things got simpler 👍🏻